numpy/lib/histograms.py

```diff
@@ -220,6 +220,14 @@ def _hist_bin_auto(x):
 def _ravel_and_check_weights(a, weights):
     """ Check a and weights have matching shapes, and ravel both """
     a = np.asarray(a)
+
+    # Ensure that the array is a "subtractable" dtype
+    if a.dtype in [bool, np.bool_]:
```

Comment:
Just use `a.dtype == np.bool_`.

Using `==` on a dtype just casts the non-dtype into a dtype too.